feat: upgrade to github action v2 (#6)
Decathlon/release-notes-generator-action: README.md

    Release Note Generator GitHub Action
  </p>
</h1>
<p align="center">
  This repository provides a GitHub action to <strong>automatically create a release notes</strong> when a milestone is closed.
</p>

**Table of Contents**
  - [Common usage](#common-usage)
  - [Why?](#why)
  - [Breaking change](#breaking-change)
  - [Startup](#startup)
    - [Use Github action](#use-github-action)
      - [Settings for v1.0.0+ release (deprecated)](#settings-for-v100-release-deprecated)
      - [Settings for v2.0.0+ release](#settings-for-v200-release)
      - [NOTE](#note)
      - [Result illustration](#result-illustration)
    - [Configure output folder](#configure-output-folder)
    - [Custom output release file](#custom-output-release-file)
      - [Prefixed name using v1.0.0+](#prefixed-name-using-v100)
      - [Prefixed name using v2.0.0+](#prefixed-name-using-v200)
    - [Use Milestone title](#use-milestone-title)
      - [Prefixed name using v1.0.0+](#prefixed-name-using-v100-1)
      - [Prefixed name using v2.0.0+](#prefixed-name-using-v200-1)
      - [Result illustration](#result-illustration-1)
  - [How to use the generated file](#how-to-use-the-generated-file)

## Common usage

> Each time a milestone will be closed, this GitHub action will scan closed issues/PR attached to the milestone, and then, create automatically a wonderful release note.  

Reviewer: ```suggestion
Each time a milestone is closed, this GitHub action scans its attached issues and pull request, and automatically generates a wonderful release note.  
```

> _The action is using the [Spring.io release-notes generator](https://github.com/spring-io/github-release-notes-generator) tool._

Reviewer: ```suggestion
_The action uses [Spring.io release-notes generator](https://github.com/spring-io/github-release-notes-generator) tool._
```

<p align="center">
  <img src="https://github.com/Decathlon/release-notes-generator-action/raw/master/images/release_notes.png" alt="Result illustration"/>
<p>

## Why?

> Why we are not simply automating the release creation and push the release notes there?  

Reviewer: ```suggestion
Why aren't we simply automating the release creation and push the release notes there?  
```

> We think in some cases it is better to keep release notes separated from the release process... yeah, maybe the right name is not release notes anymore. :) Taking as example a continuous deployment project: __anytime you are pushing to the repository a new version of the application is built and pushed in production.__ Ahd so, a new release is created to complete the process; a release with a **single** function inside.  

Reviewer: ```suggestion
We think in some cases it is better to keep release notes separated from the release process... yeah, maybe the right name is not release notes anymore. :) Taking as example a continuous deployment project: __anytime you push to the repository, a new version of the application is built and pushed in production.__ And so, a new release is created to complete the process; a release with a **single** feature inside.  
```

> We think it is not a good idea to push to your 'application users' a new release notes message anytime you're pushing (let us say 10 times a day?).  

Reviewer: ```suggestion
We think it is not a good idea to push to 'users' a new release notes message every time you generate a changes in the source code (let us say 10 times a day?).  
```

> In this such of projects we prefer to communicate only once a Sprint (or with the time split you prefer), which gave you a bigger release notes containing the information pushed in production with the latest 30/40 releases.

Reviewer: ```suggestion
It may be a better way to communicate only once every week or once the sprint is over, which gives you a clearer and complete release notes containing the aggregated information pushed in production within the last sprint (or any days).
```

## Breaking change

> Starting from August 2019, GitHub switch [Actions syntax from HCL to YAML](https://help.github.com/en/articles/migrating-github-actions-from-hcl-syntax-to-yaml-syntax).  

Reviewer: ```suggestion
Starting from August 2019, GitHub team switch [Actions syntax from HCL to YAML](https://help.github.com/en/articles/migrating-github-actions-from-hcl-syntax-to-yaml-syntax).  
```
